src/Components/Analyzers/src/ComponentParameterAnalyzer.cs

```diff
@@ -61,13 +62,20 @@ public override void Initialize(AnalysisContext context)
                     {
                         var captureUnmatchedValuesParameters = new List<IPropertySymbol>();
 
-                            // Per-property validations
-                            foreach (var property in properties)
+                        // Per-property validations
+                        foreach (var property in properties)
                         {
-                            if (property.SetMethod?.DeclaredAccessibility == Accessibility.Public)
+                            if (property.DeclaredAccessibility != Accessibility.Public)
                             {
                                 context.ReportDiagnostic(Diagnostic.Create(
-                                    DiagnosticDescriptors.ComponentParametersShouldNotBePublic,
+                                    DiagnosticDescriptors.ComponentParametersShouldBePublic,
+                                    property.Locations[0],
```

Comment:
I'm never really sure about doing this. Is it possible for Locations to be empty? Would `FirstOrDefault` be better here?

Reply:
It's what we do above but can definitely do `FirstOrDefault` to be extra sure.